modules/rostests/apitests/user32/GetDCEx.c

```diff
@@ -466,16 +567,748 @@ Test_GetDCEx_CS_SwitchedStyle()
 
     DestroyWindow(hwnd1);
     DestroyWindow(hwnd2);
-    ok(UnregisterClass(pszClassName, GetModuleHandleA(0)) == TRUE,
+    ok(UnregisterClassW(pszClassName, GetModuleHandleA(0)) == TRUE,
        "UnregisterClass failed\n");
 }
 
+/* test behavior of DC attributes with various GetDC/ReleaseDC combinations */
+static void test_dc_attributes(void)
+{
+    HDC hdc, old_hdc;
+    HDC hdcs[20];
+    INT i, rop, def_rop, caps;
+    BOOL found_dc;
+
+    /* test cache DC */
+
+    hdc = GetDC( hwnd_cache );
+    def_rop = GetROP2( hdc );
+
+    SetROP2( hdc, R2_WHITE );
+    rop = GetROP2( hdc );
+    ok( rop == R2_WHITE, "wrong ROP2 %d\n", rop );
+
+    ok( WindowFromDC( hdc ) == hwnd_cache, "wrong window\n" );
+    ReleaseDC( hwnd_cache, hdc );
+    ok( WindowFromDC( hdc ) == 0, "wrong window\n" );
+    hdc = GetDC( hwnd_cache );
+    rop = GetROP2( hdc );
+    ok( rop == def_rop, "wrong ROP2 %d after release\n", rop );
+    SetROP2( hdc, R2_WHITE );
+    ok( WindowFromDC( hdc ) == hwnd_cache, "wrong window\n" );
+    ReleaseDC( hwnd_cache, hdc );
+    old_hdc = hdc;
+
+    found_dc = FALSE;
+    for (i = 0; i < 20; i++)
+    {
+        hdc = hdcs[i] = GetDCEx( hwnd_cache, 0, DCX_USESTYLE | DCX_NORESETATTRS );
+        if (!hdc) break;
+        rop = GetROP2( hdc );
+        ok( rop == def_rop, "wrong ROP2 %d after release %p/%p\n", rop, old_hdc, hdc );
+        if (hdc == old_hdc)
+        {
+            found_dc = TRUE;
+            SetROP2( hdc, R2_WHITE );
+        }
+    }
+    if (!found_dc)
+    {
+        trace( "hdc %p not found in cache using %p\n", old_hdc, hdcs[0] );
+        old_hdc = hdcs[0];
+        SetROP2( old_hdc, R2_WHITE );
+    }
+    while (i > 0) ReleaseDC( hwnd_cache, hdcs[--i] );
+
+    for (i = 0; i < 20; i++)
+    {
+        hdc = hdcs[i] = GetDCEx( hwnd_cache, 0, DCX_USESTYLE | DCX_NORESETATTRS );
+        if (!hdc) break;
+        rop = GetROP2( hdc );
```

Comment:
We do typically not space-pad that. I have seen more than one location in your PR where that undesired style was used.
```suggestion
        rop = GetROP2(hdc);
```